granulate_utils/containers/cri.py

```diff
@@ -73,57 +53,46 @@ def _reconstruct_name(container: Union[api_pb2.Container, api_pb2.ContainerStatu
         return "_".join(["k8s", container_name, sandbox_name, namespace, sandbox_uid, restart_count])
 
     def list_containers(self, all_info: bool) -> List[Container]:
-        containers: List[Container] = []
-
-        for rt, path in self._runtimes.items():
-            with RuntimeServiceWrapper(path) as stub:
-                for container in stub.ListContainers(api_pb2.ListContainersRequest()).containers:
-                    if all_info:
-                        # need verbose=True to get the info which contains the PID
-                        status_response = self._container_status_request(stub, container.id, verbose=True)
-                        if status_response is None:
-                            # container probably went down
-                            continue
-                        pid: Optional[int] = json.loads(status_response.info.get("info", "{}")).get("pid")
-                        containers.append(self._create_container(status_response.status, pid, rt))
-                    else:
-                        containers.append(self._create_container(container, None, rt))
-
+        containers = []
+        with self.stub() as stub:
+            for container in stub.ListContainers(self.api.api_pb2.ListContainersRequest()).containers:
+                if all_info:
+                    # need verbose=True to get the info which contains the PID
+                    status_response = self._container_status_request(stub, container.id, verbose=True)
+                    if status_response is None:
+                        # container probably went down
+                        continue
+                    pid: Optional[int] = json.loads(status_response.info.get("info", "{}")).get("pid")
+                    containers.append(self._create_container(status_response.status, pid))
```

Comment:
nitpick - these 2 lines can be shared as a utility function with `get_container` which has the same logic.